### ISSUE TITLE
Discrepancy in the Number of Queries between Object Detection and Affordances

### ISSUE THREAD

**Issue by anh0001** [anh0001/affomernet#5]
**Description**: 
There is an error in the current implementation where the number of queries between object detection and affordances might be different. This can lead to inconsistencies during the matching process. 

Please check the file `rtdetr_decoder.py`, as the affordances queries might return more queries than expected.

**Steps to Reproduce**:
1. Run the model with the current configuration.
2. Observe the number of queries returned for object detection and affordances.

**Expected Behavior**:
The number of queries for object detection and affordances should be consistent.

**Actual Behavior**:
The number of queries for affordances is higher than that for object detection.

**File to Check**: 
- [`rtdetr_decoder.py`](https://github.com/anh0001/affomernet/blob/main/src/zoo/rtdetr/rtdetr_decoder.py)

**Comment by anh0001** [comment]
# Fix Query Count Mismatch in RTDETR Decoder

## Issue #5 - Query Count Inconsistency Between Detection and Affordances

### Fix Description
This addresses the query count mismatch between object detection and affordances in the RTDETR decoder. The issue was causing inconsistencies during the matching process due to improper handling of denoising queries splitting.

### Changes Made
Added query splitting logic in the `forward` function of `rtdetr_decoder.py` to properly handle denoising queries when training:

```python
# In rtdetr_decoder.py - forward function
if self.training and dn_meta is not None:
    split_size = dn_meta['dn_num_split']
    dn_out_bboxes, out_bboxes = torch.split(out_bboxes, split_size, dim=2)
    dn_out_logits, out_logits = torch.split(out_logits, split_size, dim=2)
    dn_features, decoder_features = torch.split(decoder_features, split_size, dim=2)
